### PR TITLE
Ci improvements

### PR DESCRIPTION
**Pull request by timotheeguerin**
- Use corepack to install pnpm: Faster and respect the pnpm version set in package.json instead of having another place to keep up to date
- dependency cache
- upgrade to new code coverage task
- Move all consitency check to independent github action workflow(Makes it easier to see which one failed immediately without having to open devops and dig into the steps)

### CODE REVIEW

File: eng/pipelines/pr.yml
Comment (markcowl) [inline eng/pipelines/pr.yml:5]: don't we want to try this in the typespec-azure repo for a while before moving here?

**Review by bterlson — APPROVED** [review]

**Review by markcowl — CHANGES_REQUESTED** [review]
just thinking we want to hold off on merge queues until we have more experience in typespec-azure

**Review by markcowl — APPROVED** [review]

File: eng/pipelines/pr.yml
Comment (timotheeguerin) [reply eng/pipelines/pr.yml:5]: this just allows it to be enabled, it doesn't hurt to have it

Comment (timotheeguerin) [reply eng/pipelines/pr.yml:5]: I am not actually changing to use merge queue in here but that will then just be a lever flip if we decide to

**Review by markcowl — CHANGES_REQUESTED** [review]
Also, do we need to make format/lint/spell check required checks now?

**Comment by timotheeguerin** [comment]
> Also, do we need to make format/lint/spell check required checks now?

yes we do I'll do that as soon as this merge

**Comment by timotheeguerin** [comment]
or I can do now I guess it doesn't hurt as longas this gets merged now

**Review by markcowl — APPROVED** [review]

**Pull request merged**
